packages/itwinui-react/src/core/LabeledSelect/LabeledSelect.tsx

```diff
@@ -150,9 +150,12 @@ export const LabeledSelect = React.forwardRef(
           disabled={disabled}
           className={className}
           style={style}
-          {...{ required: props.native ? required : undefined }}
-          {...rest}
           ref={forwardedRef}
+          {...({
+            required: props.native ? required : undefined,
+            ...rest,
+          } as SelectProps<T>)}
```

Comment:
What is going on here?

Reply:
When doing `Omit<SelectProps<T>, "styleType">` instead of `SelectProps<T>`, I was getting strange type behaviors and errors. This type assertion fixes (ignores) the error. But open to proper ways to fix it:

<details>
<summary>Error screenshots</summary>

<img width="382" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/382bd704-e203-4bb8-89b4-ffad0c302e07">
<img width="584" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/ba898000-fdfa-42e2-a431-c2c7f4bf6d6b">

</details>

Reply:
After our recent commits (https://github.com/iTwin/iTwinUI/pull/1886/files/ae07a6fb42b4d464e50d42a3cd0347dc31c43da0..aca5cf10a4c6b7504e9eae54b630913074789f41), that type assertion was removed.